app/models/dish.rb

```diff
@@ -14,10 +14,12 @@ def self.add_ingredients(dish,ingredients)
     end
     
     def self.add_ingredients_with_dictionary(dish,ingredient_dictionaries)
-        ingredient_dictionaries.each do |index,item|
-            ingredient = Ingredient.create(name: item[:name], portion_size: item[:amount] )
-            dish.ingredients.append(ingredient)
-            ingredient.dish = dish 
+        if ingredient_dictionaries != nil
+            ingredient_dictionaries.each do |index,item|
+                ingredient = Ingredient.create(name: item[:name], portion_size: item[:amount] )
```

Comment:
What happens to the ingredients that are created here if the Dish they're part of doesn't get saved?